uuid.go

```diff
@@ -220,6 +243,47 @@ func (u UUID) MarshalText() (text []byte, err error) {
 	return
 }
 
+// UnmarshalJSON implements the json.Unmarshaler interface.
+// It works the same as UnmarshalText except the quotation marks are removed if present.
+func (u *UUID) UnmarshalJSON(data []byte) (err error) {
+	if len(data) < 32 {
+		err = fmt.Errorf("uuid: UUID string too short: %s", data)
+		return
+	}
+
+	t := data[:]
+
+	if bytes.Equal(t[:9], urnPrefix) {
+		t = t[9:]
+	} else if t[0] == '{' || t[0] == '"' {
```

Comment:
You are right. I'll fix that bug.

No, becase you can recieve a byte slice without the leading ". I think (will check for sure) the default go's json parser send a string without the ", because it handles the uuid properly, so you have to check. But for sure we can reuse the same code for both methods.